2025/lab/Keys/Week7_Missingness_Robustness.Rmd

```diff
@@ -0,0 +1,551 @@
+  ) %>%
+  select(-hrs1, -conrinc, -prestg10, -hispanic) 
+
+```
+
+In past week's we've run `drop_na()` on our dataset to ensure we only have observations with no missingness. This week, we're going to leave those cases in.
+
+## Handling Missing Data
+
+### Complete-Case Analysis
+
+Complete-case analysis is where you drop observations which are missing on any variable in the dataset, even if you're not using them. This is never really a good idea because you lose data points that didn't need to be tossed.
+
+```{r gss-complete case}
+
+gss_comp <- gss %>% drop_na()
+
+```
+
+### Available-Case Analysis
+
+This is what we have done in this course thus far. Available-case analysis is where you drop observations which have missing values on any of the variables you are working with *in a specific model*. This is also called "**listwise deletion**." This requires you somewhat anticipate the model specifications before hand (though you can always come back--that's what's great about coding in the source pane). 
+
+```{r available-case analysis}
+
+gss_avail <- gss %>% 
+  select(logincome, prestige, self_employed, race, female, wtssnrps) %>% drop_na()
+
+```
+
+
+## Descriptives
+
+As always, let's look at our descriptive statistics. This time, pay attention to the number of missing observations for each variable.
+
+```{r desc-tables}
+
+datasummary_skim(gss,
+                 type = "numeric",
+                 fmt = 2, # Show 2 decimal places 
+                 histogram = T,
+                 title = "Sample Descriptive Statistics: Continuous Variables",
+                 notes = "Data: 2022 General Social Survey",
+                 output = "kableExtra")
+
+datasummary_skim(gss, 
+                 type = "categorical",
+                 title = "Sample Descriptive Statistics: Categorical Variables",
+                 notes = "Data: 2022 General Social Survey",
+                 output = "huxtable")
+
+```
+
+Unfortunately, this function doesn't give us missingness on categorical variables. What we can do instead is run the below code, which will create a category `NA` for missing values.
+
+```{r tidy-cat}
+
+gss_cat_table <- gss %>% 
+  select(where(~ !is.numeric(.))) %>%  # Select columns that are NOT numeric
+  pivot_longer(cols = everything(), names_to = "variable", values_to = "value") %>% 
+  group_by(variable, value) %>% 
+  summarise(n = n(), .groups = "drop") %>%  # Counting number of values within each variable 
+  mutate(percent = round(n / sum(n), 2)) %>%
+  ungroup() %>%
+  as.data.frame()
+
+gss_cat_table
+
+```
+
+# Some Models
+
+Now that we know what our data looks like, and know there are missing values, let's run some models. The models use the same predictors, but the second model (`mod2`) only uses observations that have no missing data across all variables and the third model (mod3) only uses observations that have no missing data across only the variables selected for in the model (`data = gsscomp`).
```

Comment:
@brenthoagland This might be a little tedious to implement, but I don't think the complete-case analysis part is very meaningful (i.e. dropping observations based on variables that are not even considered). I think the lab could be streamlined by just dropping this part and just including a sentence above on complete-case analysis.